### PR TITLE
Correctness fixes for GPU flipping and interrupts

### PR DESCRIPTION
**Pull request by yuriks**
Please see individual commit messages.

This fixes the blinking issue present in several games. (See last commit.)

The hwtest mentioned is at https://github.com/yuriks/3ds-gputest/tree/master/flipping

### CODE REVIEW

**Comment by yuriks** [comment]
Hold off on this, I decided to investigate some more.

File: src/video_core/renderer_opengl/renderer_opengl.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ void RendererOpenGL::LoadFBToActiveGLTexture(const GPU::Regs::FramebufferConfig&
     // TODO: Why are active_fb and the valid framebuffer flipped compared to 3dbrew documentation
```
Comment (archshift) [inline src/video_core/renderer_opengl/renderer_opengl.cpp]: Comment no longer applies.

**Comment by yuriks** [comment]
Managed to fix the issue. Ready for review. I'd like to request testing in games which had flickering problems before to see if those have been completely resolved.

**Comment by Subv** [comment]
LGTM

**Comment by neobrain** [comment]
Good to see this code finally being fixed up, given that we basically just hacked this up until stuff worked and then left it untouched :) 

LGTM.

**Comment by bunnei** [comment]
Tested a lot of homebrew and retail stuff. Works really great, LGTM!!

**Pull request merged**